augur/data/schema-multiplot.json

```diff
@@ -0,0 +1,126 @@
+{
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "$id": "https://nextstrain.org/schemas/multiplot/v1",
+    "title": "Nextstrain multiplot data schema proposal",
+    "type": "object",
+    "additionalProperties": false,
+    "required": ["config", "collections"],
+    "properties": {
+        "config": {
+            "description": "Config for display text and display options",
+            "type": "object",
+            "additionalProperties": false,
+            "required": ["collections", "groupings"],
+            "properties": {
+                "collections": {
+                    "description": "The available collection options and their display titles",
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "description": "A single collection option",
+                        "type": "object",
+                        "additionalProperties": false,
+                        "required": ["key"],
+                        "properties": {
+                            "key": {
+                                "description": "The property name of the collection",
+                                "type": "string"
+                            },
+                            "title": {
+                                "description": "The title to display in the collections dropdown. Optional -- if not provided, then `key` will be used",
+                                "type": "string"
+                            }
+                        }
+                    }
+                },
+                "groupings": {
+                    "description": "The available group by options and their display titles",
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "description": "A single group by option",
+                        "type": "object",
+                        "additionalProperties": false,
+                        "required": ["key"],
+                        "properties": {
```

Comment:
Some groupings will have a link back to the tree - for instance, `serum` will often be tips in the tree (think cogs in nextflu), `serum_clade` will often be a branch label etc. While not part of the initial prototype, we may one day implement behavior where interacting with an individual multi-plot / facet highlights the associated node / part of the tree. In this case, we could add a property here to indicate what we should search for in the tree. E.g.

```json
{
    "key": "serum",
    "value": "Reference serum",
    "tree_link": "name",
},
{
    "key": "serum_clade",
    "value": "Clade of reference sera tested against",
    "tree_link": "branch_label::clade"
}
```



(This is all rather vague of course, just thinking through things to make sure we allow for this direction.)

Reply:
This is definitely a cool future direction to keep open! We would want to think through how to expose all of the available `tree_links` for a dataset.

Reply:
Yeah - there are some really interesting ideas here, for instance defining a grouping based on mutations (e.g. glycosylation sites) and seeing where in the tree they map to, but they're further down the road for sure.